vim/vimrc

```diff
@@ -1,204 +1,149 @@
-scriptencoding utf-8
-
-" needs to be first line
-set nocompatible
-filetype off
-set rtp+=~/.vim/bundle/Vundle.vim
-call vundle#begin()
-
-Plugin 'gmarik/Vundle.vim'
-
-Plugin 'kchmck/vim-coffee-script'
-Plugin 'kien/ctrlp.vim'
-Plugin 'tpope/vim-endwise'
-Plugin 'tpope/vim-fugitive'
-Plugin 'tpope/vim-haml'
-Plugin 'tpope/vim-markdown'
-Plugin 'tpope/vim-rails'
-Plugin 'tpope/vim-surround'
-Plugin 'tpope/vim-repeat'
-Plugin 'scrooloose/syntastic'
-Plugin 'nathanaelkane/vim-indent-guides'
-Plugin 'SirVer/ultisnips'
-Plugin 'godlygeek/tabular'
-Plugin 'majutsushi/tagbar'
-Plugin 'vim-ruby/vim-ruby'
-Plugin 'Lokaltog/vim-powerline'
-Plugin 'vim-scripts/nginx.vim'
-Plugin 'vim-scripts/AutoTag'
-Plugin 'vim-scripts/LargeFile'
-Plugin 'mattn/emmet-vim'
-Plugin 'pangloss/vim-javascript'
-Plugin 'aalvarado/ultisnips-snippets.git'
-Plugin 'docunext/closetag.vim'
-Plugin 'nono/vim-handlebars'
-Plugin 'briancollins/vim-jst'
-Plugin 'jinfield/vim-nginx'
-Plugin 'sjbach/lusty'
-Plugin 'ervandew/supertab.git'
-Plugin 'scrooloose/nerdcommenter'
-Plugin 'AndrewRadev/vim-eco'
-Plugin 'mxw/vim-jsx'
-
-" Themes
-Plugin 'altercation/vim-colors-solarized'
-Plugin 'larssmit/vim-getafe'
-Plugin 'wesgibbs/vim-irblack'
-
-call vundle#end()
-filetype plugin indent on
-
-" Because we care for your eyes
-set background=dark
-
-:colorscheme koehler
-
 if filereadable(expand("~/.vimrc.before"))
   source ~/.vimrc.before
 endif
 
-set hidden
-set number
-set ts=2 sts=2 sw=2 expandtab
-syntax on
-" Because it never works well
-setglobal nobomb
-
-" Lots of history
-set history=1000
-
-" Don't include hidden buffers in autocomplete
-set complete=.,b,u,t
-
-" Make cmdline tab completion similar to bash
-set wildmode=list:longest
-
-" I'm pretty sure this works for everyone
-set term=screen-256color
-
-" Shorter messages
-set shortmess+=I
-
-if has('mouse')
-  set mouse=a
+if empty(glob('~/.vim/autoload/plug.vim'))
+  silent !curl -fLo ~/.vim/autoload/plug.vim --create-dirs
```

Comment:
Not sure how to go about this since it's hard to say, but this assumes curl is pre-installed, otherwise the plugin system won't work, this may work for some OS'es but not others, maybe it is indeed safer to do do this in the `install` shell script and if curl is not present ask the user to install it, and in the `Rakefile` use `httprb` to make sure this step wont fail?

Perhaps I'm missing something, what do you think? @aalvarado

Reply:
ah, that would work, yeah